src/arrayMethodSort.js

```diff
@@ -4,8 +4,25 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    let isSwapped;
+
+    do {
+      isSwapped = false;
+
+      for (let i = 0; i < this.length; i++) {
+        const condition = compareFunction
+          ? compareFunction(this[i], this[i + 1]) > 0
+          : String(this[i]) > String(this[i + 1]);
+
+        if (condition) {
+          [this[i], this[i + 1]] = [this[i + 1], this[i]];
+          isSwapped = true;
+        }
+      }
```

Comment:
The loop condition in the for loop should be 'i < this.length - 1' instead of 'i < this.length'. Without the '-1', when 'i' is at the last index, 'this[i + 1]' will be 'undefined', and the comparison will not behave correctly.